src/Route.php

```diff
@@ -170,10 +174,30 @@ public function defaults(array $defaults): self
 
     /**
      * @param callable|MiddlewareInterface $middleware
-     * @return MiddlewareInterface
+     * @return MiddlewareInterface|string|array
      */
-    private function prepareMiddleware($middleware): MiddlewareInterface
+    private function prepareMiddleware($middleware)
     {
+        if (
+            is_string($middleware) && is_subclass_of($middleware, MiddlewareInterface::class)
+        ) {
+            if ($this->container === null) {
+                throw new InvalidArgumentException('Route container must not be null for lazy loaded middleware.');
+            }
+            return $middleware;
+        }
+
+        if (
+            is_array($middleware) && count($middleware) === 2
+            && isset($middleware[0], $middleware[1])
+            && is_string($middleware[1]) && is_string($middleware[0]) && class_exists($middleware[0])
```

Comment:
```suggestion
            is_string(reset($middleware)) && is_string(next($middleware)) && class_exists(reset($middleware))
```
At now we can check any indexes, but I don't like this solution :(

Reply:
See tests